network/http/source/main.c

```diff
@@ -5,41 +5,133 @@
 
 #include <3ds.h>
 
-Result http_download(httpcContext *context)//This error handling needs updated with proper text printing once ctrulib itself supports that.
+Result http_download(char *url)
 {
 	Result ret=0;
+	httpcContext context;
+	char *newurl=NULL;
 	u8* framebuf_top;
 	u32 statuscode=0;
-	u32 size=0, contentsize=0;
-	u8 *buf;
+	u32 contentsize=0, readsize=0, size=0;
+	u8 *buf, *lastbuf;
 
-	ret = httpcBeginRequest(context);
-	if(ret!=0)return ret;
+	printf("Downloading %s\n",url);
+	gfxFlushBuffers();
 
-	ret = httpcGetResponseStatusCode(context, &statuscode, 0);
-	if(ret!=0)return ret;
+	do {
+		ret = httpcOpenContext(&context, HTTPC_METHOD_GET, url, 1);
+		printf("return from httpcOpenContext: %"PRId32"\n",ret);
+		gfxFlushBuffers();
 
-	if(statuscode!=200)return -2;
+		// This disables SSL cert verification, so https:// will be usable
+		ret = httpcSetSSLOpt(&context, SSLCOPT_DisableVerify);
+		printf("return from httpcSetSSLOpt: %"PRId32"\n",ret);
+		gfxFlushBuffers();
 
-	ret=httpcGetDownloadSizeState(context, NULL, &contentsize);
-	if(ret!=0)return ret;
+		// Enable Keep-Alive connections (on by default, pending ctrulib merge)
+		// ret = httpcSetKeepAlive(&context, HTTPC_KEEPALIVE_ENABLED);
+		// printf("return from httpcSetKeepAlive: %"PRId32"\n",ret);
+		// gfxFlushBuffers();
 
-	printf("size: %"PRId32"\n",contentsize);
-	gfxFlushBuffers();
+		// Set a User-Agent header so websites can identify your application
+		ret = httpcAddRequestHeaderField(&context, (char*)"User-Agent", (char*)"httpc-example/1.0.0");
```

Comment:
iirc, the c++ compiler bitched continually about them until I cast them char*, and I've just copy/pasted from that code. I'll clean things up in a push in a bit here since I have to fix a memory leak with the redirection malloc() still.